agent/api/ecsclient/client.go

```diff
@@ -212,11 +213,17 @@ func getCpuAndMemory() (int64, int64) {
 	return int64(cpu), mem
 }
 
-func getAdditionalAttributes() []*ecs.Attribute {
-	return []*ecs.Attribute{&ecs.Attribute{
+func (client *APIECSClient) getAdditionalAttributes() []*ecs.Attribute {
+	attribs := []*ecs.Attribute{&ecs.Attribute{
 		Name:  aws.String("ecs.os-type"),
 		Value: aws.String(api.OSType),
 	}}
+	for attribute, _ := range client.config.InstanceAttributes {
+		attribs = append(attribs, &ecs.Attribute{
+			Name: aws.String(attribute),
+			Value: aws.String(client.config.InstanceAttributes[attribute])})
```

Comment:
Why not:

```go
for attribute, value := range client.config.InstanceAttributes {
	attributes = append(attributes, &ecs.Attribute{
		Name:  aws.String(attribute),
		Value: aws.String(value),
	}
}
```

You don't need to do `client.config.InstanceAttributes[attribute]` when you can just get that from iterating over the map.